add roadmap 2021.12 (#301)
oam-dev/kubevela.io: docs/roadmap/2021-12-roadmap.md

---
title:  Roadmap
---

Date: 2021-10-01 to 2021-12-30

## Core Platform

> 1. Support multi cluster rollout and multi version deployment.

Reviewer: ```suggestion
1. Support multi-cluster rollout and multi-version deployment.
```

> 2. Support more cloud resources provision with the same OAM spec.

Reviewer: ```suggestion
2. Improve cloud resource provisioning support on Alibaba and AWS.
```

> 3. Support application level observability, log and metrics collection and show in grafana.

Reviewer: ```suggestion
3. Support application observability: collect and show log and metrics in Grafana.
```

## User Experience

> 1. Release Dashboard for deploying KubeVela Application.

Reviewer: ```suggestion
1. Enhance Dashboard to improve KubeVela experience on app delivery workflow, lifecycle management, addon center, etc.
```

> 2. Release APIServer for PaaS integration and usage.

Reviewer: ```suggestion
2. Release APIServer for REST API integration.
```

> 3. Align vela CLI with the new features such as workflow and env.

Reviewer: ```suggestion
3. Align vela CLI with the new features: workflow and env.
```

> 4. Support vela doctor command to help diagnose application.

Reviewer: ```suggestion
4. Support vela doctor command to help diagnose user issues and collect bug trace.
```

## Third-party integrations

> 1. Integrate with OpenYurt for edge computing.

Reviewer: ```suggestion
1. Provide OpenYurt integration to improve IoT/edge experience.
```

> 2. Integrate with ArgoCD for GitOps.

Reviewer: ```suggestion
2. Provide ArgoCD integration to improve GitOps experience.
```
